map/download-preplanned-map/src/main/java/com/esri/samples/download_preplanned_map/DownloadPreplannedMapController.java

```diff
@@ -0,0 +1,415 @@
+              showWebMapButton.setDisable(false);
+              downloadAreaButton.setDisable(false);
+              deleteOfflineAreasButton.setDisable(false);
+
+              // display error details if the job fails
+            } else if (downloadPreplannedOfflineMapJob.getStatus() == Job.Status.FAILED) {
+              new Alert(Alert.AlertType.ERROR, "Download Preplanned Offline Map Job failed. Error: " + downloadPreplannedOfflineMapJob.getError());
+            }
+          });
+        } catch (InterruptedException | ExecutionException e) {
+          new Alert(Alert.AlertType.ERROR, "Could not create Default Parameters for the Download Preplanned Offline Map Job.").show();
+        }
+      });
+    } else {
+      new Alert(Alert.AlertType.ERROR, "No Preplanned Map Area selected for downloading.").show();
+    }
+  }
+
+  /**
+   * Checks for layer and table errors of an offline map result, and displays them if present.
+   *
+   * @param downloadPreplannedOfflineMapResult the result to query for errors
+   */
+  private void checkForOfflineMapResultErrors(DownloadPreplannedOfflineMapResult downloadPreplannedOfflineMapResult) {
+    if (downloadPreplannedOfflineMapResult.hasErrors()) {
+      // accumulate all layer and table errors into a single message
+      StringBuilder stringBuilder = new StringBuilder("Errors: ");
+
+      Map<Layer, ArcGISRuntimeException> layerErrors = downloadPreplannedOfflineMapResult.getLayerErrors();
+      layerErrors.forEach((layer, exception) ->
+              stringBuilder.append("Layer: ").append(layer.getName()).append(". Exception: ").append(exception.getMessage()).append(". ")
+      );
+
+      Map<FeatureTable, ArcGISRuntimeException> tableError = downloadPreplannedOfflineMapResult.getTableErrors();
+      tableError.forEach((table, exception) ->
+              stringBuilder.append("Table: ").append(table.getTableName()).append(". Exception: ").append(exception.getMessage()).append(". ")
+      );
+
+      // show the message
+      new Alert(Alert.AlertType.ERROR, "One or more errors occurred with the Offline Map Result: " + stringBuilder.toString()).show();
+    }
+  }
+
+  /**
+   * Resets the map view to show the web map from the portal.
+   */
+  @FXML
+  private void showWebMap() {
+    // set the web map to the map vew
+    mapView.setMap(webMap);
+
+    // show the graphics overlay with the areas of interest
+    areasOfInterestGraphicsOverlay.setVisible(true);
+
+    // add the listener to the list view to change the viewpoint on selection
+    preplannedAreasListView.getSelectionModel().selectedItemProperty().addListener(listViewSelectionChangeViewpointListener);
+
+    // disable the 'view web map' button
+    showWebMapButton.setDisable(true);
+  }
+
+  /**
+   * Deletes all the previously downloaded offline areas.
+   */
+  @FXML
+  private void deleteOfflineAreas() {
+    try {
+      // cancel the job
+      if (downloadPreplannedOfflineMapJob != null) {
```

Comment:
I took over this block from the .NET sample. Agree that with the way it's set up, it's not possible to trigger this method because the button is disabled, and I think it was the same in .NET. Perhaps it's simply advocating good practice, that any potential job should be stopped and cancelled?

Reply:
As discussed, removing this as it seems unnecessary